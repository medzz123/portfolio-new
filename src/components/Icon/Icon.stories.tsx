import { Meta, Story } from '@storybook/react/types-6-0';
import React from 'react';

import Icon from './Icon';

export default {
  title: 'Components/Icon',
  component: Icon,
} as Meta;

const Template: Story = (storyArguments) => <Icon name="sun" {...storyArguments} />;

export const Default = Template.bind({});
