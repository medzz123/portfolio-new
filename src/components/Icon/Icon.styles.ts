import { styled } from '@theme/config';

export const IconContainer = styled('i', {});
