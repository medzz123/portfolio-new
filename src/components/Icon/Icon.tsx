import React, { FunctionComponent } from 'react';

import { IconProps } from './Icon.models';
import { IconContainer } from './Icon.styles';

const Icon: FunctionComponent<IconProps> = (props) => {
  const { name, size = 1 } = props;

  return <IconContainer css={{ '--ggs': size }} className={`gg-${name}`} />;
};

export default Icon;
