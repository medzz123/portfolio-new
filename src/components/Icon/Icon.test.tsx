import { render } from '@test/library';
import React from 'react';

import Icon from './Icon';

test('Icon Test', () => {
  const { container } = render(<Icon name="enter" />);

  expect(container).toMatchSnapshot();
});
