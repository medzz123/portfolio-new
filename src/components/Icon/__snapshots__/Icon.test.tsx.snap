// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Test 1`] = `
<div>
  <i
    class="sx03kze sx03kze-uzmqd gg-enter"
  />
</div>
`;
